lib/jekyll/tags/post_url.rb

```diff
@@ -18,7 +18,7 @@ def ==(other)
       end
 
       def deprecated_equality(other)
-        date = Utils.parse_date(name, "'#{name}' does not contain valid date and/or title.")
+        date = Utils.parse_date(@date, "'#{@date}' does not contain valid date.")
```

Comment:
I don't even know why we're doing this because `date` is an `attr_reader` for `@date` anyway so why we're shadowing it instead of just parsing it in the initializer?